dev/components/notifications/component.js

```diff
@@ -36,15 +34,19 @@
 
     function getNotifications () {
       notificationsService.getNotifications().then( function(res) {
+
         vm.notifications = res.data;
 
-        vm.notifications.map( function (notification) {
-          return notificationsService.messageGenerator(notification);
-        });
+        if (vm.notifications !== []){
```

Comment:
This will never work
Arrays in JS are objects which work by references so you try to compare two objects

Reply:
What problem are you trying to resolve?

Reply:
Using `if` increasing code complexity. We should avoid code branching where it possible. Please remove it, it resolves nothing